crates/lib3h/src/dht/dht_protocol.rs

```diff
@@ -2,29 +2,109 @@ use crate::dht::PeerAddress;
+use lib3h_ghost_actor::prelude::*;
+use lib3h_protocol::data_types::*;
+
 pub type FromPeerAddress = PeerAddress;
 
-#[derive(Debug, PartialEq, Clone)]
-pub enum DhtCommand {
-    /// Owner received a gossip bundle from a remote peer, and asks us to handle it.
+pub type DhtFactory = fn(config: &DhtConfig) -> Lib3hResult<Box<DhtActor>>;
+
+pub type DhtActor = dyn GhostActor<
+    DhtRequestToParent,
+    DhtRequestToParentResponse,
+    DhtRequestToChild,
+    DhtRequestToChildResponse,
+    Lib3hError,
+>;
+pub type DhtEndpointWithContext<UserData> = GhostContextEndpoint<
+    UserData,
+    DhtContext,
+    DhtRequestToParent,
+    DhtRequestToParentResponse,
+    DhtRequestToChild,
+    DhtRequestToChildResponse,
+    Lib3hError,
+>;
+pub type DhtEndpoint = GhostEndpoint<
+    DhtRequestToChild,
+    DhtRequestToChildResponse,
+    DhtRequestToParent,
+    DhtRequestToParentResponse,
+    Lib3hError,
+>;
+pub type ChildDhtWrapperDyn<UserData> = GhostParentWrapperDyn<
+    UserData,
+    DhtContext,
+    DhtRequestToParent,
+    DhtRequestToParentResponse,
+    DhtRequestToChild,
+    DhtRequestToChildResponse,
+    Lib3hError,
+>;
+
+pub type DhtToChildMessage =
+    GhostMessage<DhtRequestToChild, DhtRequestToParent, DhtRequestToChildResponse, Lib3hError>;
+
+pub type DhtToParentMessage =
+    GhostMessage<DhtRequestToParent, DhtRequestToChild, DhtRequestToParentResponse, Lib3hError>;
+
+#[derive(Debug)]
+pub enum DhtContext {
+    NoOp,
+    RequestAspectsOf {
+        entry_address: Address,
+        aspect_address_list: Vec<Address>,
+        msg: EntryListData,
+        request_id: String,
+    },
+    RequestEntry(DhtToChildMessage),
+    QueryEntry(QueryEntryData),
+}
+
+#[derive(Debug)]
+pub enum DhtRequestToChild {
+    /// Commands
+    /// Parent received a gossip bundle from a remote peer, and asks us to handle it.
     HandleGossip(RemoteGossipBundleData),
-    /// Owner wants a specific entry.
-    FetchEntry(FetchDhtEntryData),
-    /// Owner wants us to hold a peer discovery data item.
+    /// Parent wants us to hold a peer discovery data item.
     HoldPeer(PeerData),
-    /// Owner notifies us that it is holding one or several Aspects for an Entry.
+    /// Parent notifies us that it is holding one or several Aspects for an Entry.
     /// Note: Need an EntryData to know the aspect addresses, but aspects' content can be empty.
     HoldEntryAspectAddress(EntryData),
-    /// Owner wants us to bookkeep an entry and broadcast it to neighbors
+    /// Parent wants us to bookkeep an entry and broadcast it to neighbors
     BroadcastEntry(EntryData),
-    /// Owner notifies us that is is not holding an entry anymore.
+    /// Parent notifies us that is is not holding an entry anymore.
     DropEntryAddress(Address),
-    /// Owner's response to ProvideEntry request
-    EntryDataResponse(FetchDhtEntryResponseData),
+
+    /// Requests
+    /// Parent wants PeerData for a specific Peer
+    RequestPeer(String),
```

Comment:
Can do we better than a `String` type here?

Reply:
Not really since its the typed used in the DHT.